src/class-rest.php

```diff
@@ -68,11 +69,18 @@ public function register_parsely_meta(): void {
 			$post_id = $object['id'];
 			$options = $this->parsely->get_options();
 			$post    = WP_Post::get_instance( $post_id );
-			return $this->parsely->construct_parsely_metadata( $options, $post );
+			$meta    = $this->parsely->construct_parsely_metadata( $options, $post );
+
+			return array(
+				'version' => self::PARSELY_REST_VERSION,
+				'meta'    => $meta,
+			);
 		};
 
-		$args = array( 'get_callback' => $callback );
-		register_rest_field( 'post', self::PARSELY_META_REST_FIELD_NAME, $args );
-		register_rest_field( 'page', self::PARSELY_META_REST_FIELD_NAME, $args );
+		$options     = $this->parsely->get_options();
+		$object_type = array_merge( $options['track_post_types'], $options['track_page_types'] );
```

Comment:
We could probably do with an `array_unique()` here, in case someone has registered the same type for both post and page (we don't block the checkboxes yet).

Also, should there be a filter here?

Reply:
> We could probably do with an array_unique() here

Agreed, adding in a commit.

> Also, should there be a filter here?

We could, I imagine that you are suggesting this in case a user wants logic more complex than ON/OFF, which the checkboxes already provide? I would add the filter in a further iteration to keep it simple at the beginning.

Reply:
More that I'm thinking an engineer may want to guarantee what REST API outputs are affected, regardless of what some colleague may fiddle with the checkboxes are accidentally break things.